app/src/main/java/com/duckduckgo/app/browser/newtab/NewTabLegacyPageView.kt

```diff
@@ -138,12 +139,15 @@ class NewTabLegacyPageView @JvmOverloads constructor(
             viewState.favourites.isEmpty()
 
         if (!showLogo && isHomeBackgroundLogoVisible) {
+            onHasContent?.invoke(false)
             this.gone()
         } else {
             this.show()
             if (isHomeBackgroundLogoVisible) {
+                onHasContent?.invoke(false)
                 homeBackgroundLogo.showLogo()
             } else {
+                onHasContent?.invoke(true)
                 homeBackgroundLogo.hideLogo()
```

Comment:
```suggestion
        if (!showLogo && isHomeBackgroundLogoVisible) {
            onHasContent?.invoke(false)
            this.gone()
        } else {
            this.show()
            onHasContent?.invoke(true)
            if (isHomeBackgroundLogoVisible) {
                homeBackgroundLogo.showLogo()
            } else {
                homeBackgroundLogo.hideLogo()
```

Wouldn't this be enough?